### PR TITLE
demos: GH862 Fix build errors of demos (cube, vulkaninfo) on the wayl…

### PR DESCRIPTION
**Pull request by elongbug**
…and only window system

Signed-off-by: Mun Gwan-gyeong <elongbug@gmail.com>

### CODE REVIEW

**Comment by elongbug** [comment]
this patch fixes #862

File: demos/cube.c [hunk 1]
```diff
@@ -29,7 +29,7 @@
 #include <stdbool.h>
 #include <assert.h>
 #include <signal.h>
-#if defined(__linux__) && !defined(ANDROID)
+#if defined(VK_USE_PLATFORM_XLIB_KHR) | defined(VK_USE_PLATFORM_XCB_KHR)
```
Comment (MarkY-LunarG) [inline demos/cube.c]: Both these should use the boolean or || not the bitwise or |.

Comment (elongbug) [inline demos/cube.c]: Thanks for your comments. I have updated with your guide.

**Comment by Tony-LunarG** [comment]
Cherry picked to master - thanks

**Pull request closed without merging**